### ISSUE TITLE
[dolmen] Support (check-sat-assuming)

### ISSUE THREAD

**Issue by bclement-ocp** [OCamlPro/alt-ergo#695]
The legacy psmt2 frontend supports `(check-sat-assuming)`, but the Dolmen frontend currently doesn't.

Or rather, it partially does — it seems to *ignore* (!!) the assumptions, see this example:

```smtlib2
(set-logic ALL)

(define-fun b () Bool false)
(check-sat-assuming (b))
```

The legacy frontend correctly answers `sat`, but the dolmen frontend answers `unknown`.

**Comment by bclement-ocp** [comment]
Surprisingly, however, the legacy frontend complains that `(check-sat-assuming not yet supported)` whereas the dolmen frontend does not complain. Go figure :shrug:

**Comment by bclement-ocp** [comment]
(Taking that one)

**Comment by bclement-ocp** [comment]
Actually fixed by #660

**Comment by bclement-ocp** [comment]
Actually, yes, but no.

```smtlib2
(set-logic ALL)
(define-fun b () Bool false)
(define-fun c () Bool true)
(check-sat-assuming (c b))
```

crashes ( #660 only added support for `check-sat-assuming` with a single assuming)
